services/redis/queue.js

```diff
@@ -59,21 +59,30 @@ const registerV2ServiceQueue = async (token, userId, follows, topics, locations,
   }
 };
 
+const setRequiredTime = (delay) => {
+  let requiredTime = momentTz().tz('America/Los_Angeles');
+  if (delay === 0) {
+    const randomTime = Math.floor(Math.random() * (40 - 10 + 1)) + 10;
+    requiredTime = momentTz().tz('America/Los_Angeles').add(randomTime, 'minutes');
+  } else {
+    const randomTime = sample([6, 7, 8]);
+    const additionalDays = delay;
+    requiredTime = momentTz()
+      .tz('America/Los_Angeles')
+      .set({hour: randomTime})
+      .add(additionalDays, 'days');
+  }
+  return requiredTime;
+};
+
 const followTopicServiceQueue = async (user_id, topic_id, community_message_format_id, delay) => {
   const data = {
     user_id,
     topic_id,
     community_message_format_id
   };
   let currentTime = momentTz().tz('America/Los_Angeles');
-  const randomTime = sample([6, 7, 8]);
-  const additionalDays = delay;
-
-  let requiredTime = momentTz()
-    .tz('America/Los_Angeles')
-    .set({hour: randomTime})
-    .add(additionalDays, 'days');
-
+  let requiredTime = setRequiredTime(delay);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `followTopicServiceQueue` function, the `currentTime` is set before the `requiredTime`. If the `setRequiredTime` function takes a significant amount of time to execute, this could introduce a small but potentially important discrepancy between these two times. To ensure accuracy, consider setting `currentTime` immediately before it's used to calculate `diffTime`.
```diff
-   let currentTime = momentTz().tz('America/Los_Angeles');
    let requiredTime = setRequiredTime(delay);
+   let currentTime = momentTz().tz('America/Los_Angeles');
```

Please note that these are suggestions based on the provided code snippets and might need adjustments based on the overall context of your application.


<!-- This is an auto-generated comment by OSS CodeRabbit -->